Misc/NEWS

```diff
@@ -1100,6 +1100,10 @@ Documentation
 Build
 -----
 
+- bpo-29941: Add ``--with-assertions`` configure flag to explicitly enable
+  C asertions. Defaults to off, but ``--with-pydebug`` also enables
```

Comment:
Adjusted the wording, but avoided mentioning NDEBUG (because it's *not* using NDEBUG that enables assertions :P). The backticks are reST, not Markdown, and the NEWS file is supposed to be reST.